### ISSUE TITLE
False positive when calling `tuple` constructor under influence of bidirectional type inference

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#7085]
The following should result in no error because of the recently-agreed-upon rules for `tuple[Any, ...]`, but pyright generates an error here because its bidirectional type inference logic "upgrades" the type of `tuple(x)` from `tuple[Any, ...]` to `tuple[int, ...]`. But the latter has different (inconsistent) semantics because of the recent rule change about `tuple[Any, ...]`.

```python
def f(x: Iterable[Any]):
    a: tuple[int, int] = tuple(x)  # Should not generate error
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.349, which I just published.
